cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/AnomalyDetectorState.java

```diff
@@ -287,12 +295,17 @@ private SelfHealingEnabledRatio selfHealingEnabledRatio() {
    * Update anomaly status once associated self-healing operation has finished.
    *
    * @param anomalyId Unique id of anomaly which triggered self-healing operation.
+   * @param completeWithError Whether the task execution finished with error.
    */
-  public synchronized void markSelfHealingFinished(String anomalyId) {
+  public synchronized void markSelfHealingFinished(String anomalyId, boolean completeWithError) {
     if (_ongoingSelfHealingAnomaly == null || !_ongoingSelfHealingAnomaly.anomalyId().equals(anomalyId)) {
       throw new IllegalStateException(String.format("Anomaly %s is not marked as %s state in AnomalyDetector.",
                                                     anomalyId, AnomalyState.Status.FIX_STARTED));
     }
+    long totalTime = _time.milliseconds() - _ongoingSelfHealingAnomaly.detectionTimeMs();
+    LOG.info("Self-healing for anomalyId {} completed{}. Total time: {} ms", anomalyId, completeWithError ? " with error" : "", totalTime);
```

Comment:
nit: it would be slightly easier to grep for successful ones if the string were "... completed successfully." for them (instead of just "... completed.")

Reply:
updated